### ISSUE TITLE
value of topo_missing is -150.d0 m, which is a reasonable value

### ISSUE THREAD

**Issue by bolliger32** [clawpack/geoclaw#327]
I'm looking in `topo_module.f90`. Why doesn't this get set to something like -99999.d0? This could be changed easily in this module source code, but I'm assuming other modules would also need to change? It's not clear to me how this value is interpreted or how missing values are interpolated.

**Comment by mandli** [comment]
At the very least we should provide a way via the Python setting to perhaps set the fill value as there are times that you probably don't care too much what the values are but setting it to something negative would be default fill that space in with water.  In reality what would be even better is to mask the areas out and depend on data from another topography file automatically.

**Comment by rjleveque** [comment]
Agreed -- I did something along these lines recently for a project where we had a file with bathymetry values only and no topography, in which case we wanted to set some positive values  in the cells missing data. 

The easiest way to modify this is to change the value set in `geoclaw/src/2d/shallow/topo_module.f90` in the line
```
        real(kind=8), parameter :: topo_missing = -150.d0
```
and then do `make new`.  That should do it, but we should really add this as a value that can be set in `setrun.py` rather than hardwired in. I'll try to do this...

Once this value has replaced the `no_data_value` read in from the topofile, the topo is then used in the usual manner to define a piecewise bilinear function that is integrated to get cell averages of the topo/bathymetry.

**Comment by bolliger32** [comment]
Interesting. To me this seems fairly problematic, depending on where that missing data is, no? Like you could be creating a big trench or mountain somewhere in the ocean. Ultimately, it seems like the best approach would be basically what @mandli said: to (a) check lower-res topography for values to fill in, and (b) raise an error if there is no lower res data to fill in.

In the meantime, I think the proper workaround for us would be to preprocess our DEMs and do that approach (fill in missing values with lower res data).

**Comment by rjleveque** [comment]
Yes, filling in would probably be better provided the coarser topo is sufficiently fine that it does something reasonable near the edges. And this would be a bit hard to implement in a general robust way, I think.

So in general it may be best to just provide a simple mechanism to fill all missing data with a single value for problems where that makes sense, and otherwise leave it to the user to preprocess the DEM as you suggest to do whatever they think is best.

But at least we need to document this better and make it easier to adjust that value, and probably a different default value would be less likely to cause subtle problems for users who aren't aware of what's going on.

Another option to force the user to be aware of this is to not allow any no_data_value's, and require the user to preprocess by replacing these with something?  But then there's the danger that they re-use their topo file later for something else and forget that it was originally missing data....

**Comment by bolliger32** [comment]
Yeah thinking about it I'd imagine it's not an easily generalizable feature. My thought is that in this case the default should probably be an error, rather than filling in with a value...and then the user could override this default and pick a number to fill in missing vals with. OR could preprocess the files and replace the missings as they see fit (which we will do).

**Comment by mandli** [comment]
We should provide some guidance on how to do this at the very least.

**Comment by mandli** [comment]
PR incoming from @rjleveque.
